packages/salesforcedx-vscode-core/src/commands/util/libraryCommandlet.ts

```diff
@@ -9,22 +9,28 @@ import * as vscode from 'vscode';
 import { channelService } from '../../channels';
 import { nls } from '../../messages';
 import { notificationService } from '../../notifications';
-import {
-  TelemetryBuilder,
-  telemetryService
-} from '../../telemetry';
+import { TelemetryBuilder, telemetryService } from '../../telemetry';
 import { CommandletExecutor } from './sfdxCommandlet';
 
-export abstract class LibraryCommandletExecutor<T>
+export interface LibraryExecution<U = void> {
+  success: boolean;
+  result?: U;
+}
+
+export abstract class LibraryCommandletExecutor<T, U = void>
   implements CommandletExecutor<T> {
   protected readonly telemetry = new TelemetryBuilder();
   protected readonly revealChannelOutput = false;
   protected abstract readonly logName: string;
   protected abstract readonly executionName: string;
 
-  protected abstract run(response: ContinueResponse<T>): Promise<boolean>;
+  protected abstract run(
```

Comment:
Library commands will now return a `LibraryExecution` object that will only include the `success` property by default. If the result object is needed from a command, then that can be specified when the individual executor is declared and it will be added as a property to the `LibraryExecution` result

Reply:
Sorry I'm not following, where is a result needed outside of the context of `run`?

Reply:
For example, we use the response from the `force:apex:log:list` command in the `parameterGatherer` for the `force:apex:log:get` command [here](https://github.com/forcedotcom/salesforcedx-vscode/blob/2757f2a94a3e8f24efd7ab9c967df62700e1b084/packages/salesforcedx-vscode-core/src/commands/forceApexLogGet.ts#L144)

Reply:
Since the libraries give us freedom from CLI shelling now, what if we used `logService.getLogRecords()` directly in the gatherer?

Reply:
Yeah we totally could do that too! I was thinking the main benefit with using the executors is that in case running `getLogRecords()` takes a long time, the user at least sees the progress bar letting them know that the command is running. Otherwise if we call `getLogRecords()` directly, they may just have a blank screen for a bit

Reply:
What about adding this to the gatherer and calling it in `gather()`?

```typescript
private async getLogRecords(): Promise<LogRecord[]> {
  const connection = await workspaceContext.getConnection();
  const logService = new LogService(connection);
  return vscode.window.withProgress({
    location: vscode.ProgressLocation.Notification,
    title: "Fetching log list"
  }, () => logService.getLogRecords();
}
```